minicluster/src/main/java/org/apache/accumulo/miniclusterImpl/MiniAccumuloClusterImpl.java

```diff
@@ -624,53 +640,80 @@ private void verifyUp() throws InterruptedException, IOException {
           "Error starting TabletServer " + tsExpectedCount + "- instance not started");
     }
 
-    var zrw = getServerContext().getZooReaderWriter();
-    String rootPath = getServerContext().getZooKeeperRoot();
+    Watcher w = new Watcher() {
+      public void process(WatchedEvent event) {
+        if (event.getState() == KeeperState.Expired) {
+          log.debug("Session expired, state of current session : {}", event.getState());
+        }
+      }
+    };
+    try (ZooKeeper zk = new ZooKeeper(getZooKeepers(), 60000, w)) {
 
-    int tsActualCount = 0;
-    int tryCount = 0;
-    try {
-      while (tsActualCount != tsExpectedCount) {
-        tryCount++;
-        tsActualCount = 0;
-        for (String child : zrw.getChildren(rootPath + Constants.ZTSERVERS)) {
-          tsActualCount++;
-          if (zrw.getChildren(rootPath + Constants.ZTSERVERS + "/" + child).isEmpty())
-            log.info("TServer " + tsActualCount + " not yet present in ZooKeeper");
+      String secret = getSiteConfiguration().get(Property.INSTANCE_SECRET);
+
+      for (int i = 0; i < numTries; i++) {
+        if (zk.getState().equals(States.CONNECTED)) {
+          zk.addAuthInfo("digest", ("accumulo" + ":" + secret).getBytes(UTF_8));
```

Comment:
We should have a static utility method for constructing the auth info, so we don't have to do this more than one place.

Reply:
Are you suggesting that something already exists for this, or that I should create it? I see where this is done in other tests. I don't see evidence of a utility.

Reply:
I'm suggesting we should have one, if we don't already. It can be done separately from this PR, though.

Reply:
Did this in #2424

Reply:
```suggestion
          ZooUtil.digestAuth(zk, secret);
```